From f9993d0dfa598c65371f3cf70d3988c77840c8c5 Mon Sep 17 00:00:00 2001
From: Claudia <claui@users.noreply.github.com>
Date: Sun, 9 Jul 2023 12:31:30 +0200
Subject: [PATCH] Fix deprecation introduced in Pillow 9.2.0

Pillow 8.0.0 introduced a `getbbox` method [1], which is meant to
replace `getsize`.

Since Pillow 9.2.0, `getsize` has been deprecated [2] [3] and subsequently
removed in Pillow 10.0.0. [4]

Transition to `getbbox` in order to fix the deprecation. This also
unblocks a future dependency version upgrade to Pillow 10.

[1]: https://pillow.readthedocs.io/en/stable/releasenotes/8.0.0.html#imagedraw-textbbox
[2]: https://pillow.readthedocs.io/en/stable/releasenotes/9.2.0.html#font-size-and-offset-methods
[3]: https://github.com/python-pillow/Pillow/blob/1953e4353874fc6da9445022adeaf41008e89201/CHANGES.rst#L521
[4]: https://pillow.readthedocs.io/en/stable/releasenotes/10.0.0.html#font-size-and-offset-methods
---
 src/dymoprint/dymo_print_engines.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/dymoprint/dymo_print_engines.py b/src/dymoprint/dymo_print_engines.py
index 6046a2e..0e38358 100644
--- a/src/dymoprint/dymo_print_engines.py
+++ b/src/dymoprint/dymo_print_engines.py
@@ -265,9 +265,9 @@ def render_text(
             frame_width = min(frame_width, 3)
 
         font = ImageFont.truetype(font_file_name, fontsize)
-        label_width = max(font.getsize(line)[0] for line in labeltext) + (
-            font_offset * 2
-        )
+        boxes = (font.getbbox(line) for line in labeltext)
+        line_widths = (right - left for left, _, right, _ in boxes)
+        label_width = max(line_widths) + (font_offset * 2)
         text_bitmap = Image.new("1", (label_width, label_height))
         with draw_image(text_bitmap) as label_draw:
             # draw frame into empty image